src/obstacle_extractor.cpp

```diff
@@ -131,6 +136,10 @@ void ObstacleExtractor::pclCallback(const sensor_msgs::PointCloud::ConstPtr pcl_
 }
 
 void ObstacleExtractor::processPoints() {
+  prev_segments_.push_front(move(segments_));
+  if (prev_segments_.size() > p_prev_seg_buffer_size_)
```

Comment:
normally we don't really use single line if statements, but it seems that this repo has that as part of its style so 👍